src/fs.rs

```diff
@@ -507,6 +507,15 @@ impl Fs {
         &self.home_directory
     }
 
+    pub fn change_current_directory(&mut self, path: &str) {
+        if path.starts_with('/') {
+            self.current_directory = GuestPathBuf::from(GuestPath::new(&path));
+        } else {
+            self.current_directory = self.current_directory.join(path);
```

Comment:
The current `GuestPath::join` implementation doesn't handle empty, `.` and `..` path components neatly. Using `resolve_path` would not only avoid that problem, but it also would let you remove the `path.starts_with('/')` check.

Reply:
Done